### PR TITLE
Added cooling.

### PR DESCRIPTION
**Pull request by tonybaltovski**
@mikepurvis Added cooling based on `cmd_vel` and charger being connected.

### CODE REVIEW

File: ridgeback_base/src/ridgeback_cooling.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    charger_disconnected_ = true;
+  }
+
+  cmd_fans_pub_.publish(cmd_fans_msg_);
+}
+void RidgebackCooling::cmdVelCallback(const geometry_msgs::Twist::ConstPtr& twist)
+{
+  if (twist->linear.x >= 0.1 ||
+      twist->linear.y >= 0.1 ||
+      twist->angular.z >= 0.4)
```
Comment (mikepurvis) [inline ridgeback_base/src/ridgeback_cooling.cpp]: Parameterize these motion thresholds, or at least make them a constant?

File: ridgeback_base/src/ridgeback_cooling.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+
+namespace ridgeback_base
+{
+
+RidgebackCooling::RidgebackCooling(ros::NodeHandle* nh) :
+  nh_(nh),
+  charger_disconnected_(true)
+{
+  cmd_fans_pub_ = nh_->advertise<ridgeback_msgs::Fans>("mcu/fans", 1);
```
Comment (mikepurvis) [inline ridgeback_base/src/ridgeback_cooling.cpp]: I'd like this changed to `mcu/cmd_fans`, but we need to coordinate that with a firmware change.

File: ridgeback_base/src/ridgeback_cooling.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  }
+  else
+  {
+    for (int i = 0; i < 6; i++)
+    {
+      cmd_fans_msg_.fans[i] = ridgeback_msgs::Fans::FAN_ON_LOW;
+    }
+  }
+
+  cmd_fans_pub_.publish(cmd_fans_msg_);
```
Comment (mikepurvis) [inline ridgeback_base/src/ridgeback_cooling.cpp:96]: This is sketchy publishing this on receipt of two different message receipt callbacks. Can you instead set a flag and then publish from a single 10Hz timer callback?

File: ridgeback_base/src/ridgeback_cooling.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+void RidgebackCooling::statusCallback(const ridgeback_msgs::Status::ConstPtr& status)
+{
+  if (status->charger_connected)
+  {
+    cmd_fans_msg_.fans[ridgeback_msgs::Fans::CHARGER_BAY_INTAKE] = ridgeback_msgs::Fans::FAN_ON_HIGH;
+    cmd_fans_msg_.fans[ridgeback_msgs::Fans::CHARGER_BAY_EXHAUST] = ridgeback_msgs::Fans::FAN_ON_HIGH;
+    charger_disconnected_ = false;
+  }
+  else if (!charger_disconnected_)
```
Comment (mikepurvis) [inline ridgeback_base/src/ridgeback_cooling.cpp:65]: Is this var just preventing writing to the output message each time? If so, not worth it— get rid of it.

**Comment by mikepurvis** [comment]
Generally looks good, one comment above and please rebase, then we'll get this in.

File: ridgeback_base/src/ridgeback_cooling.cpp (hunk 4 above)
Comment (tonybaltovski) [inline ridgeback_base/src/ridgeback_cooling.cpp:65]: It is for the charger state logic.

**Pull request merged**